CIME/XML/env_batch.py

```diff
@@ -257,7 +258,31 @@ def make_batch_script(self, input_template, job, case, outfile=None):
             subgroup=job,
             overrides=overrides,
         )
-        output_name = get_batch_script_for_job(job) if outfile is None else outfile
+        env_workflow = case.get_env("workflow")
+
+        hidden = env_workflow.get_value("hidden", subgroup=job)
+        # case.st_archive is not hidden for backward compatibility
+        if (
+            (job != "case.st_archive" and hidden is None)
+            or hidden == "True"
+            or hidden == "true"
+        ):
+            self._hidden_batch_script[job] = True
+        else:
+            self._hidden_batch_script[job] = False
```

Comment:
How about

```py3
self._hidden_batch_script[job] = ((hidden is None and job != "case.st_archive") or (hidden is not None and hidden.lower() == "true"))
```

(The `hidden is not None` is needed because `AttributeError: 'NoneType' object has no attribute 'lower'`)
instead of the if statement?